src/vcpkg/install.cpp

```diff
@@ -740,30 +777,27 @@ namespace vcpkg::Install
                     else
                         Strings::append(msg, "    find_package(", library_target_pair.first, " CONFIG REQUIRED)\n");
 
-                    std::sort(library_target_pair.second.begin(),
-                              library_target_pair.second.end(),
-                              [](const std::string& l, const std::string& r) {
-                                  if (l.size() < r.size()) return true;
-                                  if (l.size() > r.size()) return false;
-                                  return l < r;
-                              });
+                    auto& targets = library_target_pair.second;
+                    Util::sort(targets, [](const std::string& l, const std::string& r) {
+                        if (l.size() < r.size()) return true;
+                        if (l.size() > r.size()) return false;
+                        return l < r;
+                    });
+                    targets.erase(std::unique(targets.begin(), targets.end()), targets.end());
 
-                    if (library_target_pair.second.size() <= 4)
+                    if (targets.size() <= 4)
                     {
-                        Strings::append(msg,
-                                        "    target_link_libraries(main PRIVATE ",
-                                        Strings::join(" ", library_target_pair.second),
-                                        ")\n\n");
+                        Strings::append(
+                            msg, "    target_link_libraries(main PRIVATE ", Strings::join(" ", targets), ")\n\n");
                     }
                     else
                     {
-                        auto omitted = library_target_pair.second.size() - 4;
-                        library_target_pair.second.erase(library_target_pair.second.begin() + 4,
-                                                         library_target_pair.second.end());
+                        auto omitted = targets.size() - 4;
+                        library_target_pair.second.erase(targets.begin() + 4, targets.end());
                         msg += Strings::format("    # Note: %zd target(s) were omitted.\n"
```

Comment:
fmtize?

Reply:
This should likely be non-localized, so I'll make it `{Locked}`.